enterprise/server/remote_execution/containers/firecracker/firecracker_test.go

```diff
@@ -769,6 +769,110 @@ func TestFirecrackerRunWithDocker(t *testing.T) {
 	assert.Equal(t, "", string(res.Stderr), "stderr should be empty")
 }
 
+func TestFirecrackerRunWithDockerOverTCP(t *testing.T) {
+	if *skipDockerTests {
+		t.Skip()
+	}
+
+	ctx := context.Background()
+	env := getTestEnv(ctx, t)
+	rootDir := testfs.MakeTempDir(t)
+	workDir := testfs.MakeDirAll(t, rootDir, "work")
+
+	path := filepath.Join(workDir, "world.txt")
+	if err := os.WriteFile(path, []byte("world"), 0660); err != nil {
+		t.Fatal(err)
+	}
+	cmd := &repb.Command{
+		Arguments: []string{"bash", "-c", `
+			set -e
+			# Discard pull output to make the output deterministic
+			docker -H tcp://0.0.0.0:2375 pull ` + busyboxImage + ` &>/dev/null
```

Comment:
if this doesn't work, might need 127.0.0.1 here - IIRC 0.0.0.0 only works as a serving address (meaning "any IP"), but could be wrong

Reply:
Done, but I still can't seem to get these tests pass - I've tried a bunch of things but they always seem to fail when setting things up (before we even get to this part) with:
```
firecracker_test.go:820: rpc error: code = Internal desc = failed to sync workspace: Firecracker VM crashed: rpc error: code = DeadlineExceeded desc = docker init timed out after 30s
```

Example: https://app.buildbuddy.io/invocation/57e4a583-bcdf-43df-a898-a5857075da56

Kind of at my wit's end on what else to try :(

Reply:
the VM logs might have some interesting dockerd output - you can get them with `--config=firecracker-debug`

Reply:
Thanks to your suggestion, setting `--tls=false` seemed to do the trick!